internal/server/handlers.go

```diff
@@ -175,11 +176,29 @@ func wrapLinkWithVerification(link, domain, verificationToken string) string {
 func (a *API) Login(c *gin.Context, r *api.LoginRequest) (*api.LoginResponse, error) {
 	rCtx := getRequestContext(c)
 
+	var onSuccess, onFailure func()
+
 	var loginMethod authn.LoginMethod
 	switch {
 	case r.AccessKey != "":
 		loginMethod = authn.NewKeyExchangeAuthentication(r.AccessKey)
 	case r.PasswordCredentials != nil:
+		if err := redis.RateOK(a.server.redis, r.PasswordCredentials.Name, 10); err != nil {
```

Comment:
I wonder if these `r.PasswordCredentials.Name` should be org scoped? I guess they are email addresses so it's not super critical, but it might be easy to miss that if we ever change the name to allow them to not be org scoped.

Reply:
Good point. I'll add that to the PR